<launch>
  <arg name="vehicle_frame_id" default="base_footprint" />
  <arg name="camera_frame_id"  default="zedX_left_link" />
  <arg name="debug"            default="false"/>
  <arg name="rviz"             default="false"/>

  <node pkg="lane_line_publisher" exec="lane_line_publisher" name="lane_line_publisher" output="screen">
    <param name="vehicle_frame_id" value="$(var vehicle_frame_id)"/>
    <param name="camera_frame_id"  value="$(var camera_frame_id)"/>
    <param name="debug"            value="$(var debug)"/>
    <param from="$(find-pkg-share lane_line_publisher)/config/lane_line_publisher.yaml"/>
    <param from="$(find-pkg-share vehicle)/config/ai_car1.yaml"/>

    <remap from="mask_image"           to="/aiformula_perception/road_detector/mask_image"/>
    <remap from="annotated_mask_image" to="/aiformula_perception/lane_line_publisher/annotated_mask_image"/>
    <remap from="lane_lines/left"      to="/aiformula_perception/lane_line_publisher/lane_lines/left"/>
    <remap from="lane_lines/right"      to="/aiformula_perception/lane_line_publisher/lane_lines/right"/>
    <remap from="lane_lines/center"      to="/aiformula_perception/lane_line_publisher/lane_lines/center"/>
  </node>

  <group if="$(var rviz)">
    <node pkg="rviz2" exec="rviz2" args="-d $(find-pkg-share lane_line_publisher)/rviz/lane_line_publisher.rviz"/>
  </group>
</launch>
